aep/0136.yaml

```
@@ -0,0 +1,65 @@
+aliases:
+  CustomOperation:
+    description: A custom method is identified by a path containing a colon (:) before the verb
+    targets:
+      - formats: ['oas2', 'oas3']
+        given:
+          # condition excludes custom methods Update Array custom operations defined in AEP-144, as different list of rules applies to them
+          - $.paths[?(@property.match(/:(?!add|remove)([^/]*)$/))]
+
+rules:
+  aep-136-http-method:
+    description: Custom methods should use POST or GET.
+    severity: warn
+    formats: ['oas2', 'oas3']
+    given: '#CustomOperation'
+    then:
+      function: schema
+      functionOptions:
+        schema:
+          type: object
+          properties:
+            post: {}
+            get: {}
+          additionalProperties: false
+
+  aep-136-no-patch-delete:
+    description: Custom methods should not use PATCH or DELETE.
+    severity: warn
+    formats: ['oas2', 'oas3']
+    given: '#CustomOperation'
+    then:
+      function: schema
+      functionOptions:
+        schema:
+          not:
+            type: object
+            properties:
+              patch: {}
+              delete: {}
+            additionalProperties: false
+
+  aep-136-get-idempotency:
+    description: GET-based custom methods must be idempotent and have no request body.
```

We really can't check for idempotency in the linter, so we should name the rule for what we are actually checking. For consistency with the other aeps, I think we should call this aep-136-http-body.
```suggestion
  aep-136-http-body:
    description: GET-based custom methods must have no request body.
```